src/core/drive/visit.ts

```diff
@@ -472,6 +477,13 @@ export class Visit implements FetchRequestDelegate {
     delete this.frame
   }
 
+  async renderPageSnapshot(snapshot: PageSnapshot, isPreview: boolean) {
+    await this.viewTransitioner.renderChange(this.view.shouldTransitionTo(snapshot), async () => {
+      await this.view.renderPage(snapshot, isPreview, this.willRender, this)
+      this.performScroll()
```

Comment:
I'd move the scrolling call outside `renderPageSnapshot`. We always call render and scroll together, but conceptually scrolling feels like a different concern than rendering.

Reply:
It's here because correcting the scroll position needs to happen within the transition animation. Otherwise it could transition to the page at the wrong scroll position, and then jump back to the correct scroll position after the animation is over.

So in that sense, I think scrolling _is_ part of rendering the snapshot, in as much as it's all about getting the right content into the right place on the screen. I could look at combining them into another concern to see if it makes that clearer. Might do that separately though.